compiler/runtime/OMRRuntimeAssumptions.cpp

```diff
@@ -40,3 +40,102 @@ void TR::PatchNOPedGuardSite::compensate(bool isSMP, uint8_t *location, uint8_t
    {
    _patchVirtualGuard(location, destination, isSMP);
    }
+
+TR::PatchSites::PatchSites(TR_PersistentMemory *pm, size_t maxSize) :
+    _maxSize(maxSize),
+    _size(0),
+    _refCount(0),
+    _patchPoints(NULL),
+    _firstLocation(NULL),
+    _lastLocation(NULL)
+   {
+   _patchPoints = (uint8_t**) pm->jitPersistentAlloc(sizeof(uint8_t*) * maxSize * 2);
+   }
+
+/**
+ * Add a patch site to the collection.
+ */
+void TR::PatchSites::add(uint8_t *location, uint8_t *destination)
+   {
+   TR_ASSERT_FATAL(_size < _maxSize, "Cannot add more patch sites, max size is %d", _maxSize);
+   _patchPoints[_size * 2] = location;
+   _patchPoints[_size * 2 + 1] = destination;
+   _size++;
+
+   if (_firstLocation == NULL || location < _firstLocation)
+      _firstLocation = location;
+   if (_lastLocation == NULL || location > _lastLocation)
+      _lastLocation = location;
+   }
+
+/**
+ * Compare this collection with another.
+ * Will return true if all locations are the same.
+ */
+bool TR::PatchSites::equals(PatchSites *other)
```

Comment:
`RuntimeAssumption::equals()` looks like it's unused. Just an observation - not to say that it should be removed in this commit, but maybe the behaviour of this method isn't all that relevant